File: stalin_sort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func stalin_sort(array []float64) []float64 {
	if len(array) == 0 {
		return array
	}

	max_num := array[0]
	var sorted []float64

	for _, values := range array {
		if values >= max_num {
			sorted = append(sorted, values)
			max_num = values
		} else {
			fmt.Println(values, "sent to Gulag")
		}
	}
	return sorted

}

func main() {
	rand.Seed(time.Now().Unix())
	var ini_arr_float []float64
	for i := 0; i < 50; i++ {
		initial_array := 100 * rand.Float64()
		ini_arr_float = append(ini_arr_float, initial_array)
	}
	fmt.Println(ini_arr_float)
	fmt.Println(stalin_sort(ini_arr_float))

}

File: stalin_sort.dart
import 'dart:math';

List stalin_sort(List arr) {
  if (arr.length == 0) {
    return arr;
  }
  var max_val = arr[0];
  var arr2 = [];
  for (var ele in arr) {
    if (ele >= max_val) {
      arr2.add(ele);
      max_val = ele;
    } else {
      print("${ele} sent to Gulag");
    }
  }
  return arr2;
}

main() {
  var rdm = new Random();
  var newa = [];
  for (var i = 0; i < 50; i++) {
    newa.add(100 * rdm.nextDouble());
  }
  print(newa);
  print(stalin_sort(newa));
}
